pinot-core/src/main/java/org/apache/pinot/core/accounting/PerQueryCPUMemAccountantFactory.java

```diff
@@ -868,8 +891,12 @@ void killAllQueries() {
        * use XX:+ExplicitGCInvokesConcurrent to avoid a full gc when system.gc is triggered
        */
       private void killMostExpensiveQuery() {
+        if (!_isThreadMemorySamplingEnabled) {
+          LOGGER.warn("But unable to kill query memory tracking is enabled");
```

Comment:
This log line looks strange - should it be something like `Unable to kill query because memory sampling is not enabled`?

Reply:
Also the condition earlier checked if both memory and CPU sampling were disabled but now only memory sampling is being checked?

Reply:
I've fixed the log message